solana/ts/lib/ntt.ts

```diff
@@ -129,115 +167,136 @@ export class NTT {
+    program: Program<NttBindings.NativeTokenTransfer<IdlVersion>>,
+    args: {
+      payer: PublicKey;
+      owner: PublicKey;
+      chain: Chain;
+      mint: PublicKey;
+      outboundLimit: bigint;
+      tokenProgram: PublicKey;
+      mode: "burning" | "locking";
+    },
+    pdas?: Pdas
+  ) {
     const mode: any =
       args.mode === "burning" ? { burning: {} } : { locking: {} };
     const chainId = toChainId(args.chain);
-    const mintInfo = await this.program.provider.connection.getAccountInfo(
-      args.mint
-    );
-    if (mintInfo === null) {
-      throw new Error(
-        "Couldn't determine token program. Mint account is null."
-      );
-    }
-    const tokenProgram = mintInfo.owner;
-    const ix = await this.program.methods
-      .initialize({ chainId, limit: args.outboundLimit, mode })
+
+    pdas = pdas ?? NTT.pdas(program.programId);
+
+    const limit = new BN(args.outboundLimit.toString());
+    return await program.methods
+      .initialize({ chainId, limit: limit, mode })
       .accountsStrict({
-        payer: args.payer.publicKey,
-        deployer: args.owner.publicKey,
-        programData: programDataAddress(this.program.programId),
-        config: this.pdas.configAccount(),
+        payer: args.payer,
+        deployer: args.owner,
+        programData: programDataAddress(program.programId),
+        config: pdas.configAccount(),
         mint: args.mint,
-        rateLimit: this.pdas.outboxRateLimitAccount(),
-        tokenProgram,
-        tokenAuthority: this.pdas.tokenAuthority(),
-        custody: await this.custodyAccountAddress(args.mint, tokenProgram),
+        rateLimit: pdas.outboxRateLimitAccount(),
+        tokenProgram: args.tokenProgram,
+        tokenAuthority: pdas.tokenAuthority(),
+        custody: await NTT.custodyAccountAddress(
+          pdas,
+          args.mint,
+          args.tokenProgram
+        ),
         bpfLoaderUpgradeableProgram: BPF_LOADER_UPGRADEABLE_PROGRAM_ID,
         associatedTokenProgram: splToken.ASSOCIATED_TOKEN_PROGRAM_ID,
         systemProgram: SystemProgram.programId,
       })
       .instruction();
-    await this.sendAndConfirmTransaction(
-      new Transaction().add(ix),
-      [args.payer, args.owner],
-      false
-    );
-    await this.initializeOrUpdateLUT({ payer: args.payer });
   }
 
   // This function should be called after each upgrade. If there's nothing to
   // do, it won't actually submit a transaction, so it's cheap to call.
-  async initializeOrUpdateLUT(args: {
-    payer: Keypair;
-  }): Promise<AddressLookupTableAccount> {
+  export async function initializeOrUpdateLUT(
+    program: Program<NttBindings.NativeTokenTransfer<IdlVersion>>,
+    config: NttBindings.Config<IdlVersion>,
+    args: {
+      payer: PublicKey;
+      wormholeId: PublicKey;
+    },
+    pdas?: Pdas
+  ) {
+    // if the program is at version 1.0.0, we don't need to initialize the LUT
+    if (program.idl.version === "1.0.0") return;
```

Comment:
we should introduce a more sophisticated way of checking versions than direct string matching. Here, I think we should check that the major version is at least 2 -- the rationale is that if there is a security fix that necessitates a patch bump, or an external deployment that adds some tag to the version (like `1.0.0+foo`) then it would be handled gracefully

Reply:
should we allow non-numeric values like `0+foo` for the patch version? instead of just bumping the patch version